File: app/src/main/java/com/zombietank/rockstar/news/data/NewsDatasource.java
package com.zombietank.rockstar.news.data;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface NewsDatasource {

    @GET("topstories.json")
    Single<List<Long>> getTopStories();

    @GET("item/{itemId}.json")
    Single<NewsArticle> getArticle(@Path("itemId") long id);
}

File: app/src/main/java/com/zombietank/rockstar/news/data/NewsDataSource.kt
package com.zombietank.rockstar.news.data

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface NewsDataSource {

    @get:GET("topstories.json")
    val topStories: Single<List<Long>>

    @GET("item/{itemId}.json")
    fun getArticle(@Path("itemId") id: Long): Single<NewsArticle>
}
